### PR TITLE
Slack distribution on conform loads

### PR DESCRIPTION
**Pull request by jeanChristopheGuillotin**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: src/main/java/com/powsybl/openloadflow/ac/DistributedSlackOnLoadOuterLoop.java [hunk 1]
```diff
@@ -58,7 +58,14 @@ protected double run(List<ParticipatingElement<LfBus>> participatingElements, in
 
             double targetP = bus.getLoadTargetP();
 
-            double newTargetP = targetP - remainingMismatch * factor;
+            double fixedActivePowed = bus.getFixedActivePower();
+            double variableActivePower = bus.getVariableActivePower();
+
+            double newVariableActivePower = variableActivePower - remainingMismatch * factor;
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/ac/DistributedSlackOnLoadOuterLoop.java:64]: You must change the way the factors are computed.

File: src/main/java/com/powsybl/openloadflow/network/impl/AbstractLfBus.java [hunk 2]
```diff
@@ -107,6 +111,16 @@ public double getTargetQ() {
         return getGenerationTargetQ() - getLoadTargetQ();
     }
 
+    @Override
+    public double getFixedActivePower() {
+        return 0;
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/network/impl/AbstractLfBus.java:116]: You must return the attribute.

File: src/main/java/com/powsybl/openloadflow/network/impl/AbstractLfBus.java [hunk 3]
```diff
@@ -107,6 +111,16 @@ public double getTargetQ() {
...
+    }
+
+    @Override
+    public double getVariableActivePower() {
+        return 0;
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/network/impl/AbstractLfBus.java:121]: You must return the attribute.

**Review by annetill — CHANGES_REQUESTED** [review]
You must change the class ````LfAbstractBus```` and build a unit test to improve this dev?

**Comment by annetill** [comment]
I will review as soon as possible. Can you see with Marco for the next issue that consists in keeping the power factor constant ? Thanks ! You can start from master branch, we will merge this PR later after review process.

File: src/main/java/com/powsybl/openloadflow/ac/DistributedSlackOnLoadOuterLoop.java [hunk 4]
```diff
@@ -74,6 +76,7 @@ protected double run(List<ParticipatingElement<LfBus>> participatingElements, in
                 }
 
                 bus.setLoadTargetP(newTargetP);
```
[the remark this answers is not in the document]
Comment (floriand-e2r) [reply src/main/java/com/powsybl/openloadflow/ac/DistributedSlackOnLoadOuterLoop.java:78]: Hello @annetill, in fact I think it is better to have the absolute value of the load fixed part. Indeed, here you need to update the ratio each time you change the load target. If you're dealing with the fixed part in absolute value, you don't have that problem

**Review by floriand-e2r — APPROVED** [review]
I don't see the benefit to use the per unit here, but it seems that it's the usual practice?

**Pull request merged**